core/src/main/java/org/apache/iceberg/hadoop/HadoopTables.java

```diff
@@ -194,8 +203,24 @@ TableOperations newTableOps(String location) {
     if (location.contains(METADATA_JSON)) {
       return new StaticTableOperations(location, new HadoopFileIO(conf));
     } else {
-      return new HadoopTableOperations(new Path(location), new HadoopFileIO(conf), conf);
+      return new HadoopTableOperations(new Path(location), new HadoopFileIO(conf), conf,
+              createOrGetLockManager(location));
+    }
+  }
+
+  private LockManager createOrGetLockManager(String location) {
+    Map<String, String> properties = Maps.newHashMap();
+    Iterator<Map.Entry<String, String>> configEntries = conf.iterator();
+    while (configEntries.hasNext()) {
+      Map.Entry<String, String> entry = configEntries.next();
+      String key = entry.getKey();
+      if (key.startsWith("iceberg.lock.")) {
```

Comment:
"iceberg." should be a public static variable because a process should import this variable to construct lock properties. And I wonder if we should use something more complex to avoid namespace collision, such as "iceberg.tables.hadoop."

Reply:
done